tests/netpyne-tests.js

```diff
@@ -1,505 +1,2565 @@
+    addConnection(test);
+  });
+  
+  casper.then(function() { // test adding a stimulus  source using UI
+    this.echo("######## Test Add stim Source Rule ######## ", "INFO");
+    addStimSource(test);
+  });
+  
+  casper.then(function() { // test adding a stimulus target using UI
+    this.echo("######## Test Add stimTarget Rule ######## ", "INFO");
+    addStimTarget(test);
+  });
+  
+  casper.then(function() { // test config 
+    this.echo("######## Test default simConfig ######## ", "INFO");
+    checkSimConfigParams(test);
+  });
+  
+  casper.then(function() {
+    this.reload(function() {
+      this.echo("reloading webpage", "INFO")
+    })
+  })
+  casper.then(function() {
+    this.waitWhileVisible('div[id="loading-spinner"]', function() {
+      this.wait(5000, function() { //test some expected HTML elements in landing page
+        this.echo("I've waited for netpyne to load.");
+        test.assertTitle("NetPyNE", "NetPyNE title is ok");
+        test.assertExists('div[id="widgetContainer"]', "NetPyNE loads the initial widgetsContainer");
+        test.assertExists('div[id="mainContainer"]', "NetPyNE loads the initial mainContainer");
+      });
+    }, null, 40000);
+  })
+
+  casper.then(function() { //test full netpyne loop using a demo project
+    this.echo("######## Running Demo ######## ", "INFO");
+    var demo = "from netpyne_ui.tests.tut3 import netParams, simConfig \n" +
+      "netpyne_geppetto.netParams=netParams \n" +
+      "netpyne_geppetto.simConfig=simConfig";
+    loadModelUsingPython(test, demo);
+  });
+
+  casper.then(function() { //test explore network tab functionality
+    this.echo("######## Test Explore Network Functionality ######## ", "INFO");
+    exploreNetwork(test);
+  });
+
+  casper.then(function() { //test simulate network tab functionality
+    this.echo("######## Test Simulate Network Functionality ######## ", "INFO");
+    simulateNetwork(test);
+  });
+
+  casper.run(function() {
+    test.done();
+  });
 });
 
 /**
  * Test existence of HTML elements expected when main landing page is reached
  */
-function testLandingPage(test){
-	casper.then(function () {		
-		test.assertExists('div[id="Populations"]', 'Populations button exists.');
-		test.assertExists('div[id="CellRules"]', "Cell rules button exists.");
-		test.assertExists('div[id="Synapses"]', "Synapses button exists.");
-		test.assertExists('div[id="Connections"]', "Connections button exists.");
-		test.assertExists('div[id="SimulationSources"]', "Connections button exists.");
-		test.assertExists('div[id="Configuration"]', "Configuration button exists.");
-		test.assertExists('button[id="defineNetwork"]', 'Define network button exists.');
-		test.assertExists('button[id="exploreNetwork"]', 'Explore network button exists.');
-		test.assertExists('button[id="simulateNetwork"]', 'Simulate network button exists.');
-	});
+function testLandingPage(test) {
+  casper.then(function() {
+    assertExist(test, "Populations", "div")
```

Comment:
Why did you change this bit?

Reply:
sometimes the casperTest fails saying that you cannot click inexistent "Populations" selector.
I standardized the way we assert things exist by first calling `this.waitUntilVisible`and then asserting it exist. Sometimes it was like this but in other parts it was not. So now the same function is use everywhere